stdlib/public/Concurrency/TaskStatus.cpp

```diff
@@ -1139,6 +1139,9 @@ static swift_task_escalateImpl(AsyncTask *task, JobPriority newPriority) {
   auto newStatus = oldStatus;
 
   while (true) {
+    // Ensure oldPriority is up to date if we retry the compare_exchange.
+    oldPriority = oldStatus.getStoredPriority();
```

Comment:
@mikeash small nit/question here – do we still need to read the priority on [L1138](https://github.com/swiftlang/swift/pull/80319/files#diff-3705d3ba23978722a8dc15e7ccd02655d018e1a2199aecba8c45e19eceead75cR1138) after this change?

Reply:
We don't, but it feels weird to leave it uninitialized and that's a good enough initial value. The optimizer will eliminate the redundant read for us anyway.